Flaky test: HostnameDistTest

### Before reporting an issue

- [X] I have searched existing issues
- [X] I have reproduced the issue with the latest release

### Area

testsuite

### Describe the bug

HostnameDistTest has failed 3 times the last few days, with different test methods failing each time, but with the same error.

I've also seen this test fail a few times during refactoring of the GitHub Action workflows, probably 3-4 times there as well the last few days.

### Version

main

### Expected behavior

Test passes always

### Actual behavior

Test failed 3 times last few days

### How to Reproduce?

```
============================================================================================================================================
Finding details for HostnameDistTest
============================================================================================================================================
https://github.com/keycloak/keycloak/actions/runs/3642404046

Quarkus Tests	Run Quarkus Integration Tests	2022-12-07T21:59:29.0084784Z -- [ERROR]   HostnameDistTest.testForceHostnamePortWhenNoProxyIsSet:63->assertFrontEndUrl:150->getServerMetadata:146 » Connect Connection refused (Connection refused)
Quarkus Tests	Run Quarkus Integration Tests	2022-12-07T21:59:29.0085699Z -- [ERROR]   HostnameDistTest.testUseDefaultPortsWhenProxyIsSet:70->assertFrontEndUrl:150->getServerMetadata:146 » Connect Connection refused (Connection refused)
--------------------------------------------------------------------------------------------------------------------------------------------
https://github.com/keycloak/keycloak/actions/runs/3633040350

Quarkus Tests	Run Quarkus Integration Tests	2022-12-06T21:29:51.6574261Z -- [ERROR]   HostnameDistTest.testInvalidRedirectUriWhenAdminNotSet:130 » Connect Connection refused (Connection refused)
--------------------------------------------------------------------------------------------------------------------------------------------

```

### Anything else?

_No response_

--- a/quarkus/CONTRIBUTING.md
+++ b/quarkus/CONTRIBUTING.md
@@ -189,6 +189,10 @@ and then execute the following command from the project root directory:
 mvn -f testsuite/integration-arquillian/pom.xml clean install -Pauth-server-quarkus -Dtest=OIDCProtocolMappersTest
 ```
 
+### Resolving DNS names when running tests
+
+In order to avoid using external services for DNS resolution, the tests are executed using a local host file by setting the `-Djdk.net.hosts.file=${project.build.testOutputDirectory}/hosts_file` system property.
+
 ## Documentation
 
 The documentation is a set of guides available from the [docs](../docs/guides/src/main/server) module. Please,

--- a/quarkus/tests/integration/pom.xml
+++ b/quarkus/tests/integration/pom.xml
@@ -104,6 +104,7 @@
                 <groupId>org.apache.maven.plugins</groupId>
                 <artifactId>maven-surefire-plugin</artifactId>
                 <configuration>
+                    <argLine>-Djdk.net.hosts.file=${project.build.testOutputDirectory}/hosts_file</argLine>
                     <systemPropertyVariables>
                         <kc.quarkus.tests.dist>${kc.quarkus.tests.dist}</kc.quarkus.tests.dist>
                     </systemPropertyVariables>

--- a/quarkus/tests/integration/src/test/resources/hosts_file
+++ b/quarkus/tests/integration/src/test/resources/hosts_file
@@ -1,0 +1,1 @@
+127.0.0.1 mykeycloak.org

